rust-src/concordium_base/src/pedersen_commitment/key.rs

```diff
@@ -66,6 +75,67 @@ impl<C: Curve> CommitmentKey<C> {
     }
 }
 
+impl<C: Curve> VecCommitmentKey<C> {
+    pub fn new(gs: Vec<C>, h: C) -> Self { VecCommitmentKey { gs, h } }
+
+    /// Commit to the given values using a freshly generated randomness, and
+    /// return the randomness that was generated.
+    pub fn commit<T>(
+        &self,
+        s: &[C::Scalar],
+        csprng: &mut T,
+    ) -> Option<(Commitment<C>, Randomness<C>)>
+    where
+        T: Rng, {
+        let r = Randomness::<C>::generate(csprng);
+        Some((self.hide(s, &r)?, r))
+    }
+
+    /// The low-level worker function that actually does the commitment.
+    pub fn hide_worker(
+        &self,
+        values: &[C::Scalar],
+        randomness: &C::Scalar,
+    ) -> Option<Commitment<C>> {
+        let mut bases = self.gs.clone();
+        bases.push(self.h);
+        let mut scalars = values.to_vec();
+        scalars.push(*randomness);
+        if scalars.len() != bases.len() {
+            return None;
+        }
```

Comment:
You could do this check upfront before cloning everything.

Reply:
Agree. Done.